scipy/spatial/transform/rotation.py

```diff
@@ -0,0 +1,874 @@
+        ----------
+        .. [1] `Euler angle definitions
+                <https://en.wikipedia.org/wiki/Euler_angles#Definition_by_intrinsic_rotations>`_
+        """
+        num_axes = len(seq)
+        if num_axes < 1 or num_axes > 3:
+            raise ValueError("Expected axis specification to be a non-empty "
+                             "string of upto 3 characters, got {}".format(seq))
+
+        intrinsic = (re.compile('^[XYZ]{1,3}$').match(seq) is not None)
+        extrinsic = (re.compile('^[xyz]{1,3}$').match(seq) is not None)
+        if not (intrinsic or extrinsic):
+            raise ValueError("Expected axes from `seq` to be from ['x', 'y', "
+                             "'z'] or ['X', 'Y', 'Z'], got {}".format(seq))
+
+        if any(seq[i] == seq[i+1] for i in range(num_axes - 1)):
+            raise ValueError("Expected consecutive axes to be different, "
+                             "got {}".format(seq))
+
+        seq = seq.lower()
+
+        angles = np.asarray(angles, dtype=float)
+        if degrees:
+            angles = np.deg2rad(angles)
+
+        is_single = False
+        # Prepare angles to have shape (num_rot, num_axes)
+        if num_axes == 1:
+            if angles.ndim == 0:
+                # (1, 1)
+                angles = angles.reshape((1, 1))
+                is_single = True
+            elif angles.ndim == 1:
+                # (N, 1)
+                angles = angles[:, None]
+            elif angles.ndim == 2 and angles.shape[-1] != 1:
+                raise ValueError("Expected `angles` parameter to have shape "
+                                 "(N, 1), got {}.".format(angles.shape))
+            elif angles.ndim > 2:
+                raise ValueError("Expected float, 1D array, or 2D array for "
+                                 "parameter `angles` corresponding to `seq`, "
+                                 "got shape {}.".format(angles.shape))
+        else:  # 2 or 3 axes
+            if angles.ndim not in [1, 2] or angles.shape[-1] != num_axes:
+                raise ValueError("Expected `angles` to be at most "
+                                 "2-dimensional with width equal to number "
+                                 "of axes specified, got {} for shape").format(
+                                 angles.shape)
+
+            if angles.ndim == 1:
+                # (1, num_axes)
+                angles = angles[None, :]
+                is_single = True
+
+        # By now angles should have shape (num_rot, num_axes)
+        # sanity check
+        if angles.ndim != 2 or angles.shape[-1] != num_axes:
+            raise ValueError("Expected angles to have shape (num_rotations, "
+                             "num_axes), got {}.".format(angles.shape))
+
+        quat = _elementary_quat_compose(seq, angles, intrinsic)
+        return cls(quat[0] if is_single else quat, normalized=True, copy=False)
+
+    def as_quat(self):
+        """Represent rotations as quaternions.
```

Comment:
I suggest to consider whether we use plural or singular forms in similar places. Probably the plural form is more natural.